### PR TITLE
Spreadsheet: Reuse save and rename actions

### PR DESCRIPTION
**Pull request by krkk**
*just some small cleanups* :)

- **Spreadsheet: Reuse save and rename actions**

  These parts of code were identical to their action counterparts.

- **Spreadsheet: Create rename action by using a GUI::CommonActions helper**

  Besides from reusing more parts from the code, this allows us to call the action using the F2 key. That is also the reason why we have to reassign `m_tab_context_menu_sheet_view` on tab change.

### CODE REVIEW

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 1]
```diff
@@ -357,8 +353,11 @@ void SpreadsheetWidget::setup_tabs(NonnullRefPtrVector<Sheet> new_sheets)
     if (first_tab_widget)
         change(*first_tab_widget);
 
-    m_tab_widget->on_change = [change = move(change)](auto& selected_widget) {
+    m_tab_widget->on_change = [change = move(change), this](auto& selected_widget) {
         change(selected_widget);
+
+        // for keyboard shortcuts and command palette
+        m_tab_context_menu_sheet_view = static_cast<SpreadsheetView&>(selected_widget);
```
Comment (elcuco) [inline Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:360]: Is it possible to set `m_tab_context_menu_sheet_view ` inside `change()` ?

Comment (krkk) [reply Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:360]: yeah, it is possible, but I don’t see a reason why we should move it there honestly.

It looks like the `change()` lambda function just moves function events from one view to another, so it looks more like a setup function (that probably should be improved) so setting the current tab there for the context menu doesn’t seem right.

**Comment by krkk** [comment]
Changes since last push: rebased and resolved conflicts

**Comment by AtkinsSJ** [comment]
This has a conflict.

**Comment by krkk** [comment]
Changes since last push:
- rebased and resolved conflicts
- pushed one commit that fixes an issue that appeared after this rebase :p

oh and thanks @AtkinsSJ for reminding me of this PR!

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp (hunk 1 above)
Comment (krkk) [reply Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp]: a small update: The `change()` function no longer exists (see #13032). Things from that function are now set immediately when making a tab.

**Pull request merged**